www/class/centreonHost.class.php

```diff
@@ -1094,11 +1094,20 @@ public function hasMacroFromHostChanged(
                 if (
                     isset($macro['macroInput_#index#'])
                     && isset($macro["macroValue_#index#"])
-                    && isset($macro['macroPassword_#index#'])
                 ) {
-                    if ($input == $macro['macroInput_#index#'] &&
-                        $macroValue[$ind] == $macro["macroValue_#index#"] &&
-                        $macroPassword[$ind] == $macro['macroPassword_#index#']
+                    if (
+                        $input == $macro['macroInput_#index#']
+                        && $macroValue[$ind] == $macro["macroValue_#index#"]
+                        && (
+                            (
+                                isset($macro['macroPassword_#index#'])
+                                && $macroPassword[$ind] == $macro['macroPassword_#index#']
+                            )
+                            || (
+                                isset($macro['macroPassword_#index#']) === false
+                                && isset($macroPassword[$ind]) === false
```

Comment:
if `$macroPassword[$ind]` can be not set, you will get a php notice at line 1104

Reply:
I fixed line 1104